dotcom-rendering/src/web/layouts/CommentLayout.tsx

```diff
@@ -297,24 +297,25 @@ export const CommentLayout = ({ CAPIArticle, NAV, format }: Props) => {
 
 	const contributionsServiceUrl = getContributionsServiceUrl(CAPIArticle);
 
+	const renderAds = !CAPIArticle.isAdFreeUser && !CAPIArticle.shouldHideAds;
```

Comment:
Like in your PR description, is it worth adding a TSDoc comment to these `renderAds` values to make it clear they don't apply to all slots?

Reply:
I'd agree with this suggestion 👍 

Just for my own understanding, is it the case that the logic for whether ads should be shown is in fact captured by `renderAds`, but there are reasons for not using this particular variable to control all ads at the moment? (i.e. I don't actually know what shadypie is, but I'm assuming it follows the same logic of being shown only if `!CAPIArticle.isAdFreeUser && !CAPIArticle.shouldHideAds;`?

Reply:
Shadypie is designed to fallback to a static subscription ask in the event that an ad blocker is detected. I think we could revisit these other slots on a case by case basis to think about if we want to render them or not. There's also some questions around how much of the commercial code we might be able to filter out (if we don't do this already). But I'd rather approach that in a separate thread so that we're being incremental here

Reply:
Nice summary, and thanks for adding the comment @oliverlloyd!